eo-runtime/src/main/java/EOorg/EOeolang/EOcollections/EOmultimap$EOrebuild.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2022 Eugene Darashkevich
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package EOorg.EOeolang.EOcollections;
+
+import org.eolang.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@SuppressWarnings("PMD.AvoidDollarSigns")
+public class EOmultimap$EOrebuild extends PhDefault {
+
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public EOmultimap$EOrebuild(final Phi sigma) {
+        super(sigma);
+        this.add("harr", new AtFree());
+        this.add("arr", new AtFree());
+        this.add("φ", new AtComposite(this, rho -> {
+            final Phi[] harr = new Dataized(rho.attr("harr").get()).take(Phi[].class);
+            final Phi[] arr = new Dataized(rho.attr("arr").get()).take(Phi[].class);
+
+            final List<Integer> hashes = new ArrayList<>();
```

Comment:
@EugeneDar I think we can constructs this empty list with the specified initial capacity. e.g. `new ArrayList<>(harr.length)`

Reply:
@Graur if we talk about asymptotics, it will be the same. However, then we would abandon 'for each' (because we would have to add indices), and this would make the code less readable